cli/azd/pkg/project/framework_service_dotnet.go

```diff
@@ -82,6 +83,12 @@ func (dp *dotnetProject) Initialize(ctx context.Context) error {
 	return nil
 }
 
+func replaceUnderscoreWithColon(key string) string {
+	// bicep takes .env and convert ":" to "__". dotnet secrets has to be converted back to ":"
+	// dotnet recognizes "__" as the hierarchy key separator for environment variables
+	return strings.ReplaceAll(key, "__", ":")
+}
```

Comment:
This is also similar to how the environment variable configuration provider handles variable names in .NET core.
https://learn.microsoft.com/en-us/dotnet/core/extensions/configuration-providers#environment-variable-configuration-provider